docs/use_cases/embeddings_on_browser.md

```diff
@@ -0,0 +1,628 @@
+
+This tutorial not only guides us in creating a practical, small-scale AI application but also enhances our understanding of vector embeddings in practical scenarios.
+
+Intrigued? Let's dive in and explore how to make it happen!
+
+# What we will be building
+
+Our component combines React, TensorFlow.js, and Material-UI to provide a user-friendly interface for generating and visualizing sentence embeddings. We will be taking a user input text. We will split it into sentences, generating vector embeddings for each using TensorFlow.js. To assess the quality of our embeddings, we will provide functionality to generate the similarity between two sentences in the form of a similarity matrix. Our similarity matrix provides an overview between all pairs and renders this as a colorful heatmap. Our component manages all the necessary state and UI logic to enable this interaction.
+
+# In a nutshell
+
+This is what we will be building:
+
+1. We will import necessary dependencies such as React, Material-UI components, TensorFlow.js, and D3 for color interpolation.
+2. The code defines a functional component named **`EmbeddingGenerator`,** a React functional component. This component represents the user interface for generating sentence embeddings and visualizing their similarity matrix.
+3. We declare various state variables using the **`useState`** hook to manage user input, loading states, and results.
+4. The **`handleSimilarityMatrix`** function toggles the display of the similarity matrix and calculates it when necessary.
+5. The **`handleGenerateEmbedding`** function is responsible for starting the sentence embedding generation process. It splits the input sentences into individual sentences and triggers the **`embeddingGenerator`** function.
+6. The **`calculateSimilarityMatrix`** function is a *memoized* function using the **`useCallback`** hook. It calculates the similarity matrix based on sentence embeddings.
+7. The **`embeddingGenerator`** function is an asynchronous function that loads the Universal Sentence Encoder model and generates sentence embeddings.
+8. We use the **`useEffect`** hook to render the similarity matrix as a colorful canvas when **`similarityMatrix`** changes.
+9. The component's return statement defines the user interface, including input fields, buttons, and result displays.
+10. The user input section includes a text area where the user can input sentences.
+11. The embeddings output section displays the generated embeddings.
+12. We provide two buttons. One generates the embeddings, and the other shows/hides the similarity matrix.
+13. The code handles loading and model-loaded states, displaying loading indicators or model-loaded messages.
+14. The similarity matrix section displays the colorful similarity matrix as a canvas when the user chooses to show it.
+
+# The model that we will be using
```

Comment:
I prefer headings that summarize the section, not just tease it - might be stylistic preference / SEO tweak

# Running a language model in the browser